python/ray/experimental/data/read_api.py

```diff
@@ -17,12 +17,30 @@
 def autoinit_ray(f):
     def wrapped(*a, **kw):
         if not ray.is_initialized():
-            ray.client.connect()
+            ray.client().connect()
         return f(*a, **kw)
 
     return wrapped
 
 
+def _parse_paths(paths: Union[str, List[str]]
+                 ) -> Tuple["pyarrow.fs.FileSystem", Union[str, List[str]]]:
+    from pyarrow import fs
+    if isinstance(paths, str):
+        return fs.FileSystem.from_uri(paths)
+    else:
+        if len(paths) == 0:
```

Comment:
This will fail not-so-gracefully if `paths` isn't a list, might be better to do something like:
```python
if not isinstance(paths, list) or any(not isinstance(p, str) for p in paths)):
    raise ValueError("paths must be a path string or a list of path strings.")
if len(paths) == 0:
    raise ValueError("paths list must be non-empty.")
```

Reply:
That sounds better to me. I assume it's list or str from the type hint. But I forgot that in python they are just hints...

Reply:
>But I forgot that in python they are just hints...

Isn't Python just the worst lol